scoretracker/tracker.py

```diff
@@ -1,95 +1,164 @@
-import os
+"""
+The main logic of the bot.
 
-import connexion
+This file implements the database and bot command parsing.
+"""
+from collections import defaultdict
 import json
+import os
+
 import requests
 
+import connexion
+
+
 BOT_ID = os.environ['BOT_ID']
 DB_FILE = 'data.json'
+POST_URL = 'https://api.groupme.com/v3/bots/post'
+
+
+def defaultdict_of_ints(init=None):
+    """
+    Constructor for defaultdict(int)
+    """
+    return defaultdict(int, init or {})
+
+
+def get_db(filename=DB_FILE):
+    """
+    Try to load a JSON file and return a default dictionary containing
+    the scores for a given person.
+
+    If the file is not found or is not a valid JSON object, an empty
+    collection is returned.
+    """
+    try:
+        with open(filename, 'r') as db:
+            data = json.load(db)
+    except (FileNotFoundError, IOError, json.JSONDecodeError):
+        data = {}
+
+    return defaultdict(defaultdict_of_ints, data)
 
 
 def increment_score(score, person, number):
-    with open(DB_FILE, 'w+') as db:
-        try:
-            data = db.read()
-            database = json.loads(data)
-        except Exception:
-            database = {}
-        try:
-            database[person][score] += number
-        except KeyError as ke:
-            if person in str(ke):
-                database[person] = {}
-                database[person][score] = number
-            elif score in str(ke):
-                database[person][score] = number
-        db.write(str(database))
-    rT = {}
-    rT['bot_id'] = BOT_ID
-    rT['text'] = 'Incremented '+score+' score by '+str(number)+' for '+person
-    requests.post('https://api.groupme.com/v3/bots/post', json.dumps(rT))
+    """
+    Increments/decrements the score for the given user by number.
+
+    @ed> /score show ron disappointment
+    @scorebot> ron:
+                   disappointment: 420
+    @ed> /score ron disappointment + 420
+    @ed> /score show ron disappointment
+    @scorebot> ron:
+                   disappointment: 840
+    """
```

Comment:
This exchange is incorrect, and I would prefer that the prompt you have looked more like GroupMe. The scorebot should return when the score is incremented, and rather than `@ed> `, how about `Ed: `